system_metrics_collector/src/system_metrics_collector/metrics_message_publisher.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
Can you please add a unit test for this class? I wouldn't expect a full mock, but you can just extend the publish method and assert that the message fields are created correctly.

Reply:
Added unit test.